### ISSUE TITLE
replacement content must be a string

### ISSUE THREAD

**Issue by zbraniecki** [rollup/rollup#1258]
I'm getting a weird error with node 7.4.0 and rollup 0.41.2:

file1.js:
```javascript
export class FileSource {
  constructor(name) {
    this.name = name;
  }

  hasFile() {
    return undefined;
  }
}
```

file2.js:
```javascript
import { FileSource } from './file1';
console.log(FileSource);
```

command: `rollup --format es ./file2.js` produces:

```
🚨   replacement content must be a string
```

**Comment by zbraniecki** [comment]
I couldn't find a way to minimize it more, but if I remove `undefined` or remove `this.name = name` lines it works.

**Comment by zbraniecki** [comment]
It does work in 0.41.1, so it seems to be a regression in 0.41.2

**Comment by arlac77** [comment]
same for me

**Comment by Rich-Harris** [comment]
Many thanks for the repro @zbraniecki

**Comment by rkirsling** [comment]
The following is sufficient for repro:
**foo.js**
```
class Foo {
    bar() {
        this.baz = 3;
    }
}

export default Foo;
```
****
```
import babel from 'rollup-plugin-babel';

export default {
    entry: 'foo.js',
    plugins: [
        babel({
            babelrc: false,
            presets: [['es2015', { modules: false }]],
            plugins: ['external-helpers']
        })
    ],
    targets: [
        {
            dest: 'build.js',
            format: 'es'
        }
    ]
};
```

**Comment by kzc** [comment]
Strange a stack trace wasn't printed for the error.

One possible fix:
```patch
--- a/src/ast/nodes/Identifier.js
+++ b/src/ast/nodes/Identifier.js
@@ -34,7 +34,7 @@ export default class Identifier extends Node {
                if ( this.declaration ) {
                        const name = this.declaration.getName( es );
                        if ( name !== this.name ) {
-                               code.overwrite( this.start, this.end, name, true );
+                               code.overwrite( this.start, this.end, '' + name, true );
 
                                // special case
                                if ( this.parent.type === 'Property' && this.parent.shorthand ) {
```

**Comment by Rich-Harris** [comment]
> Strange a stack trace wasn't printed for the error.

Yeah, that's a problem. For 'expected' errors (i.e. ones that are deliberately thrown, like duplicate imports) a stack kind of just clutters up the terminal, but this doesn't fall into that category. Wonder if we should print a stack for all errors that don't have a `code` property, since that's a reasonable proxy for whether or not it's expected.

Getting back to the actual bug, I think it comes from [here](https://github.com/rollup/rollup/blob/master/src/ast/nodes/MemberExpression.js#L34) — it's trying to resolve namespaced declarations (like `foo.bar` where `foo` is an imported namespace) and doesn't know how to handle keypaths that begin with `this`, like `this.bar`.

**Comment by kzc** [comment]
Rich, my fix patches the *symptom* - it's the actual place where the error was thrown. The Identifier was literally `undefined`, which as we know is a normal variable in ECMAScript, not a built-in.

**Comment by unional** [comment]
Seems like it breaks many builds. Maybe release a patch version 0.41.3 reverting to 0.41.1?

**Comment by Rich-Harris** [comment]
Released 0.41.3 with the fix. Thanks everyone, sorry for the mess!
